### PR TITLE
added two python questions, one on recursion and other on file handling.

### PR DESCRIPTION
**Pull request by rajeev033**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

### CODE REVIEW

File: src/data/python-quiz.ts [hunk 1]
```diff
@@ -1004,6 +1004,24 @@ const pythonQuiz = [
     Explanation: "If you don't specify the start index, the default start index of 0 is used.",
     Link: "https://www.freecodecamp.org/news/python-range-function-explained-with-code-examples/",
   },
+  {
+    Question: "Which data structure is used for implementation of recursive algorithm?",
```
Comment (jdwilkin4) [inline src/data/python-quiz.ts:1008]:
```suggestion
    Question: "Which data structure is used for the implementation of a recursive algorithm?",
```

File: src/data/python-quiz.ts [hunk 2]
```diff
@@ -1004,6 +1004,24 @@ const pythonQuiz = [
...
+    Answer: "Stack",
+    Distractor1: "Queue",
+    Distractor2: "Stack",
```
Comment (jdwilkin4) [inline src/data/python-quiz.ts:1011]: Distractors can't be the same as the correct answer
Please choose another option.

File: src/data/python-quiz.ts [hunk 3]
```diff
@@ -1004,6 +1004,24 @@ const pythonQuiz = [
...
+    Distractor3: "Tree",
+    Explanation: "In recursion the nested function that called later returns the value first, which is the same nature as that of a stack (Last In First Out).",
```
Comment (jdwilkin4) [inline src/data/python-quiz.ts:1013]:
```suggestion
    Explanation: "In recursion, the nested function that is called later returns the value first, which is in the same nature as that of a stack (Last In First Out).",
```

File: src/data/python-quiz.ts [hunk 4]
```diff
@@ -1004,6 +1004,24 @@ const pythonQuiz = [
...
+    Link: "https://www.freecodecamp.org/news/how-recursion-works-explained-with-flowcharts-and-a-video-de61f40cb7f9/",
```
Comment (jdwilkin4) [inline src/data/python-quiz.ts:1014]: This question on recursion is good but it should be moved over to the CS quiz.

File: src/data/python-quiz.ts [hunk 5]
```diff
@@ -1004,6 +1004,24 @@ const pythonQuiz = [
...
+  },
+  {
+    Question: "In python, which function is used to read single line from a file?",
```
Comment (jdwilkin4) [inline src/data/python-quiz.ts:1017]:
```suggestion
    Question: "In Python, which function is used to read a single line from a file?",
```

File: src/data/python-quiz.ts [hunk 6]
```diff
@@ -1004,6 +1004,24 @@ const pythonQuiz = [
...
+    Answer: "readline()",
+    Distractor1: "readlines()",
+    Distractor2: "readable()",
+    Distractor3: "readline()",
```
Comment (jdwilkin4) [inline src/data/python-quiz.ts:1021]: This distractor is the same as the correct one. 
Distractors should only be for the wrong answers.
Please choose another option here.

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

**Comment by rajeev033** [comment]
I have updated the branch with all the requested changes.

**Comment by jdwilkin4** [comment]
Thank you for making those changes.
In the future, you don't need to close the PR and open a new on when changes are requested.
You can push up your changes to the original PR and the maintainers will still be able to review it 👍

**Review by jdwilkin4 — APPROVED** [review]
Hi ! 

Congrats on your first contribution to our repository. 🎉
We look forward to future contributions. 👍

**Comment by rajeev033** [comment]
Thank You for your valuable feedback.

**Pull request merged**